packages/indexer-native/lib/index.d.ts

```diff
@@ -0,0 +1,38 @@
+/*
+    Verifies messages are signed by the same address.
+    This API may seem strange compared to the relatively straightforward alternative:
+       verify(address, message, signature): Promise<boolean>.
+    
+    But, making the API this way allows for some optimizations:
+       * The address is not repeatedly marshalled across the Rust/JS boundary
+       * The PublicKey to Address conversion is cached, saving a keccak hash
+*/
+export class NativeSignatureVerifier {
+    constructor(address: string)
+    /*
+        Verifies that the message was signed by the address this verifier
+        was constructed with.
+    */
+    async verify(message: string, signature: string): Promise<boolean>
+}
+
+export type Attestation = {
+    requestCID: string,
+    responseCID: string,
+    subgraphDeploymentID: string,
+    v: number,
+    r: string,
+    s: string,
+}
+
+/*
+    Signs attestations using the given signing data
+*/
+export class NativeAttestationSigner {
+    constructor(chainId: string | number, disputeManagerAddress: string, privateKey: string, subgraphDeploymentId: string)
```

Comment:
Why is `chainId` `string | number`? Is the string for a hex representation? I don't think we have those anywhere. It's always e.g. `"rinkeby"` or `4` and I doubt you accept `"rinkeby"` here.

Reply:
This is because U256 is supported (decimal)

Reply:
Oh, right. It's fine, I was just wondering if we could simplify call sites to just pass in a regular integer, because I haven't seen any large chain IDs out there. But this is correct, so we shouldn't change it. :)

Reply:
You already can pass in a regular integer (eg: 1)